src/coreclr/jit/lower.cpp

```diff
@@ -8777,8 +8777,8 @@ void Lowering::LowerBlockStoreAsHelperCall(GenTreeBlk* blkNode)
 #ifndef TARGET_XARCH
```

Comment:
Is there a reason to have this `#if` here still @VSadov (I assume it was here before because there was no store barrier)? Wouldn't these barriers (which are emitted as no-ops on xarch anyway) be needed for volatile effects to work properly?

Reply:
Yes, I think the `#ifndef` can be removed. 

I do not think the instructions that set up a helper call can move around in later stages. We would need barrier nodes to prevent that on x64. 
Let's do the same thing on all architectures here. Just add a comment that on XARCH the half-barriers have only optimization-inhibiting effects and do not emit anything.